### PR TITLE
Node 2452/3.6/direct connection

### PR DESCRIPTION
**Pull request by reggi**
## Description

**What changed?**

**Are there any files to ignore?**

### CODE REVIEW

File: lib/core/uri_parser.js [hunk 1]
```diff
@@ -680,11 +680,22 @@ function parseConnectionString(uri, options, callback) {
     return callback(new MongoParseError('No hostname or hostnames provided in connection string'));
   }
 
-  if (hosts.length !== 1 && parsedOptions.directConnection) {
-    // If the option is set to true, the driver MUST validate that there is exactly one host given in the host list in the URI, and fail client creation otherwise.
+  const directConnection = !!parsedOptions.directConnection;
```
Comment (reggi) [inline lib/core/uri_parser.js:683]: Curious: why use `!!` here?

File: lib/core/sdam/topology_description.js [hunk 2]
```diff
@@ -273,16 +273,23 @@ class TopologyDescription {
   }
 }
 
-function topologyTypeForServerType(serverType, options) {
-  if (serverType === ServerType.Standalone) return TopologyType.Single;
-  if (
-    options.directConnection === false &&
-    (serverType === ServerType.Unknown || serverType === ServerType.RSGhost)
-  ) {
+function topologyTypeForServerType(serverType) {
```
Comment (reggi) [inline lib/core/sdam/topology_description.js:276]: Very interesting you don't need to pass `options` in here, you can just derive from `ServerType`.

File: lib/core/uri_parser.js (hunk 1 above)
Comment (mbroadst) [reply lib/core/uri_parser.js:683]: just to ensure its a boolean, it's not technically necessary

**Comment by mbroadst** [comment]
@reggi lets go ahead and merge this, the beta goes out tonight

**Pull request merged**